docker-compose.yaml

```diff
@@ -141,16 +142,34 @@ services:
     depends_on:
       - keycloak
 
-  keycloak: # https://hub.docker.com/r/jboss/keycloak
+  postgres: # https://hub.docker.com/_/postgres
+    container_name: keycloak-postgres
+    image: postgres:${POSTGRES_VERSION:-14.5-alpine}
+    environment:
+      POSTGRES_DB: bitnami_keycloak
+      POSTGRES_USER: bn_keycloak
+      POSTGRES_PASSWORD: "#3$]O?4RGj)DE7Z!9SA5"
+
+  keycloak: # https://hub.docker.com/r/bitnami/keycloak
+    depends_on:
+      - postgres
     container_name: keycloak
-    image: jboss/keycloak:${KEYCLOAK_VERSION:-16.1.1}
+    image: bitnami/keycloak:${KEYCLOAK_SERVER_VERSION:-16.1.1}
+    volumes:
+      - ./keycloak/themes/identity:/opt/bitnami/keycloak/themes/identity
     ports:
       - "18080:8080"
     environment:
-      KEYCLOAK_USER: admin
-      KEYCLOAK_PASSWORD: admin
-    volumes:
-      - "./.keycloak/themes/identity:/opt/jboss/keycloak/themes/identity"
+      KEYCLOAK_DATABASE_HOST: keycloak-postgres
+      KEYCLOAK_DATABASE_PASSWORD: "#3$]O?4RGj)DE7Z!9SA5"
```

Comment:
This password is using `$` which is used for variable interpolation, so either escape it with another `$` or replace the `$` with another character.